### PR TITLE
update csgo egg

### PR DESCRIPTION
**Pull request by SanjaySRocks**
Added few variables for command line
Tested and works fine...

### CODE REVIEW

File: stock_eggs/source-engine/egg-counter--strike--global-offensive.json [hunk 1]
```diff
@@ -17,8 +22,8 @@
...
-            "container": "ubuntu:18.04",
+            "script": "#!\/bin\/bash\n# CSGO Installation Script\n#\n# Server Files: \/mnt\/server\napt -y update\napt -y --no-install-recommends install curl lib32gcc1 ca-certificates\n\ncd \/tmp\ncurl -sSL -o steamcmd.tar.gz http:\/\/media.steampowered.com\/installer\/steamcmd_linux.tar.gz\n\nmkdir -p \/mnt\/server\/steamcmd\ntar -xzvf steamcmd.tar.gz -C \/mnt\/server\/steamcmd\ncd \/mnt\/server\/steamcmd\n\n# SteamCMD fails otherwise for some reason, even running as root.\n# This is changed at the end of the install process anyways.\nchown -R root:root \/mnt\n\nexport HOME=\/mnt\/server\n.\/steamcmd.sh +login anonymous +force_install_dir \/mnt\/server +app_update 740 +quit\n\nmkdir -p \/mnt\/server\/.steam\/sdk32\ncp -v linux32\/steamclient.so ..\/.steam\/sdk32\/steamclient.so",
+            "container": "ubuntu:16.04",
```
Comment (parkervcp) [inline stock_eggs/source-engine/egg-counter--strike--global-offensive.json:26]: This is a downgrade.

File: stock_eggs/source-engine/egg-counter--strike--global-offensive.json [hunk 2]
```diff
@@ -34,12 +39,12 @@
...
         {
             "name": "Steam Account Token",
-            "description": "The Steam Account Token required for the server to be displayed publicly.",
+            "description": "The Steam Account Token required for the server to be displayed publicly.\r\nhttps:\/\/steamcommunity.com\/dev\/managegameservers",
             "env_variable": "STEAM_ACC",
             "default_value": "",
             "user_viewable": true,
             "user_editable": true,
-            "rules": "required|string|alpha_num|size:32"
+            "rules": "nullable|alpha_num|size:32"
```
Comment (parkervcp) [inline stock_eggs/source-engine/egg-counter--strike--global-offensive.json:47]: If you want your server to show as online this needs to be set.

File: stock_eggs/source-engine/egg-counter--strike--global-offensive.json [hunk 3]
```diff
@@ -49,6 +54,96 @@
...
+            "name": "Ticrate",
+            "description": "Set server ticrate to 64 or 128",
+            "env_variable": "TICRATE",
+            "default_value": "128",
+            "user_viewable": true,
+            "user_editable": false,
+            "rules": "required|integer|in:64,128"
+        },
+        {
+            "name": "Player Slots",
```
Comment (parkervcp) [inline stock_eggs/source-engine/egg-counter--strike--global-offensive.json:68]: I don't support limiting player slots in any way. If a company wants to add a limit they can.

File: stock_eggs/source-engine/egg-counter--strike--global-offensive.json [hunk 4]
```diff
@@ -49,6 +54,96 @@
...
+            "description": "Game Mode: 0= Casual, 1 = Competitive, 2 = Wingman",
+            "env_variable": "GAMEMODE",
+            "default_value": "1",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|integer|digits_between:0,2"
+        },
+        {
+            "name": "Map Group",
+            "description": "Set server map group: mg_active for competetive",
```
Comment (parkervcp) [inline stock_eggs/source-engine/egg-counter--strike--global-offensive.json:96]: Are there other groups that could be added to the list here?

File: stock_eggs/source-engine/egg-counter--strike--global-offensive.json [hunk 5]
```diff
@@ -1,14 +1,19 @@
...
     "author": "support@pterodactyl.io",
     "description": "Counter-Strike: Global Offensive is a multiplayer first-person shooter video game developed by Hidden Path Entertainment and Valve Corporation.",
-    "image": "quay.io\/pterodactyl\/core:source",
-    "startup": ".\/srcds_run -game csgo -console -port {{SERVER_PORT}} +ip 0.0.0.0 +map {{SRCDS_MAP}} -strictportbind -norestart +sv_setsteamaccount {{STEAM_ACC}}",
+    "features": null,
+    "images": [
+        "quay.io\/pterodactyl\/core:source"
+    ],
+    "file_denylist": [],
+    "startup": ".\/srcds_run -game csgo -console -strictportbind -usercon +ip 0.0.0.0 -port {{SERVER_PORT}} +clientport 27005 +tv_port {{TV_PORT}} +sv_setsteamaccount {{STEAM_ACC}} -maxplayers_override {{MAXPLAYERS}} -tickrate {{TICRATE}} +map {{SRCDS_MAP}} +mapgroup {{MAPGROUP}} +game_type {{GAMETYPE}} +game_mode {{GAMEMODE}} +host_workshop_collection {{WS_COLLECTIONID}} +workshop_start_map {{WS_MAP}} -authkey {{WS_APIKEY}} -nobreakpad -autoupdate -steam_dir steamcmd -steamcmd_script update.txt",
```
Comment (parkervcp) [inline stock_eggs/source-engine/egg-counter--strike--global-offensive.json:16]: Forcing the client port is a no-go.

Comment (parkervcp) [reply stock_eggs/source-engine/egg-counter--strike--global-offensive.json:16]: If a variable is not set will it error out as the next value will be another flag.

**Review by parkervcp — CHANGES_REQUESTED** [review]

**Comment by gOOvER** [comment]
It's the 2nd PR from that guy and it looks like, he have not really knowledge

File: stock_eggs/source-engine/egg-counter--strike--global-offensive.json [hunk 6]
```diff
@@ -1,14 +1,19 @@
...
-    "name": "Counter-Strike: Global Offensive",
+    "exported_at": "2022-05-03T19:37:43+05:30",
+    "name": "csgo",
     "author": "support@pterodactyl.io",
     "description": "Counter-Strike: Global Offensive is a multiplayer first-person shooter video game developed by Hidden Path Entertainment and Valve Corporation.",
-    "image": "quay.io\/pterodactyl\/core:source",
-    "startup": ".\/srcds_run -game csgo -console -port {{SERVER_PORT}} +ip 0.0.0.0 +map {{SRCDS_MAP}} -strictportbind -norestart +sv_setsteamaccount {{STEAM_ACC}}",
+    "features": null,
+    "images": [
+        "quay.io\/pterodactyl\/core:source"
```
Comment (gOOvER) [inline stock_eggs/source-engine/egg-counter--strike--global-offensive.json:13]: Really old image. Use ghcr insteed

**Comment by Software-Noob** [comment]
Without looking further into what exactly is changed, you should use the up to date csgo egg shipped with the panel as the foundation for your variables

File: stock_eggs/source-engine/egg-counter--strike--global-offensive.json (hunk 2 above)
Comment (SanjaySRocks) [reply stock_eggs/source-engine/egg-counter--strike--global-offensive.json:47]: yea, I though it should be client preference so they can use their token if they want server to be online.

File: stock_eggs/source-engine/egg-counter--strike--global-offensive.json (hunk 4 above)
Comment (SanjaySRocks) [reply stock_eggs/source-engine/egg-counter--strike--global-offensive.json:96]: yes, If its there ill add in next.

**Comment by SanjaySRocks** [comment]
Okay, will keep in mind in next update.

File: stock_eggs/source-engine/egg-counter--strike--global-offensive.json (hunk 6 above)
Comment (SanjaySRocks) [reply stock_eggs/source-engine/egg-counter--strike--global-offensive.json:13]: what image should be used?

**Comment by parkervcp** [comment]
Closing due to lack of updates.

**Pull request closed without merging**